src/main/java/com/exadel/discount/repository/VendorRepository.java

```diff
@@ -36,4 +36,7 @@
     boolean existsByIdWithNoDiscounts(@Param("vendorId") UUID vendorId);
 
     boolean existsByIdAndArchived(UUID id, boolean archived);
+
+    @EntityGraph(attributePaths = {"vendorLocations", "vendorLocations.city", "vendorLocations.city.country"})
+    Vendor save(Vendor vendor);
```

Comment:
It seems verything works good without adding this method. Check it please

Reply:
9 queries without Entity
1 query with Entity